lib/data_update_scripts/20201019192035_backfill_registered_for_users.rb

```diff
@@ -0,0 +1,12 @@
+module DataUpdateScripts
+  class BackfillRegisteredForUsers
+    def run
+      users = User.where(registered: false).or(User.where(registered_at: nil))
+      users.find_each do |user|
+        next if user.invitation_accepted_at.nil?
```

Comment:
Instead of using a guard clause I'd recommend not even fetching these users from the DB:

```ruby
users = User.where.not(invitation_accepted_at: nil).
  where(registered: false).or(User.where(registered_at: nil))

users.find_each do |user|
  user.update_columns(registered: true, registered_at: user.invitation_accepted_at)
end
```

Reply:
One step further since those `or`'s always throw me off
```ruby
users = User.where.not(invitation_accepted_at: nil).where(registered: [false, nil])
```

Reply:
Great suggestions, @citizen428 and @mstruve! I made this change.

Reply:
@mstruve The `or` seemed necessary though, the condition checked two different columns, `registered` (a boolean) and `registered_at` (a timestamp).

Reply:
You're correct @citizen428 -- I went with your implementation rather than using the `or` (even though it was a really nice way of setting it all up, had it worked) for that reason. 😊